ddtrace/contrib/asgi/middleware.py

```diff
@@ -123,6 +125,9 @@ async def __call__(self, scope, receive, send):
         # set component tag equal to name of integration
         span.set_tag_str("component", self.integration_config.integration_name)
 
+        # set span.kind to the type of request being performed
+        span.set_tag_str(SPAN_KIND, SpanKind.SERVER)
```

Comment:
if we set this here, then we probably don't need to set it in the span modifier for any integration which uses this as it's base.

Same goes for component tag since we are using the passed in integration_config (I am not 100% confident on this component part, but seems like it?)

Reply:
Yeah you're correct, Django and Starlette use this class for middleware spans. Currently we are already using this class for setting the component and span.kind tags. None of that work is happening in the patches for middleware spans specifically.